### ISSUE TITLE
Consider instrumenting the collectors and record collection errors with dedicated metrics.

### ISSUE THREAD

**Issue by stefanotorresi** [SUSE/saptune_exporter#5]
We need to ensure that `saptune` is available in the specified path.

Feel free to copy-paste from here:
https://github.com/ClusterLabs/ha_cluster_exporter/blob/3c305da4b9af7c24a8fb98389f9db5360733752c/collector/default_collector.go#L77

**Comment by MalloZup** [comment]
https://github.com/SUSE/saptune_exporter/blob/5482d0c804cd04fb412a6bca80187f916805b51d/default_collector.go#L63

this is already implemented in mostly all places unless i'm missing something?

**Comment by stefanotorresi** [comment]
Right! I totally missed that! I only checked in the collector constructors, but I see that the check happens in the actual collection cycles.
Maybe we should add the [instrumented collector](https://github.com/ClusterLabs/ha_cluster_exporter/blob/a84d5361d2d6fa85bddf8bf1673f146fe6a04d28/collector/instrumented_collector.go), then, so that we expose collection failures as well, instead of just logging the error?

**Comment by MalloZup** [comment]
make sense to add it. there are other things I didn't ported like error handling with context and maybe other details for a skeleton
